apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/logging/core/JsonLogger.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.apm.agent.core.logging.core;
+
+import com.google.gson.Gson;
+import org.apache.skywalking.apm.agent.core.logging.core.converters.LiteralConverter;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class JsonLogger extends AbstractLogger {
+    private final Gson gson;
+
+    public JsonLogger(Class<?> targetClass, Gson gson) {
+        this(targetClass.getSimpleName(), gson);
+    }
+
+    public JsonLogger(String targetClass, Gson gson) {
+        super(targetClass);
+        this.gson = gson;
+        for (Map.Entry<String, Class<? extends Converter>> entry : DEFAULT_CONVERTER_MAP.entrySet()) {
+            try {
+                if (converters instanceof LiteralConverter) {
+                    continue;
+                }
+                converters.add(entry.getValue().newInstance());
+            } catch (IllegalAccessException | InstantiationException ignore) {
```

Comment:
Why you hide the expression? What happens once these happen?

Reply:
Normally this would not happen if and only if someone change the constructor of the `Converter`s to private or so.

It will just iterate over the next `Converter` without having side effect.

Reply:
But eating the exception is not expected in the core.

Reply:
I see. Then re-thorwing another RuntimeException may be better here.

Reply:
Fixed